src/axom/quest/detail/PointFinder.hpp

```diff
@@ -208,28 +202,51 @@ class PointFinder
         countsPtr[i] = currCount;
       });
 
+    // Temporary host arrays we copy device-side data into when the candidate
+    // search is conducted on the GPU
     HostPointArray ptsHost, outIsoparHost;
     HostIndexArray outCellIdsHost;
+    HostIndexArray candidatesHost, offsetsHost, countsHost;
 
     // For sequential/OpenMP execution, just use the argument pointers
     // directly.
-    ConstHostPointView ptsHostPtr = pts;
     HostIndexView outCellIdsPtr(outCellIds, pts.size());
     HostPointView outIsoparPtr(outIsoparametricCoords, pts.size());
 
+    // If the candidate search takes place on the GPU, we need to copy the
+    // device-side data first, then set these array views to point to the
+    // intermediate arrays. Otherwise, we can set these to point to the result
+    // arrays directly.
+    ConstHostPointView ptsHostPtr;
+    HostIndexView candidatesHostPtr, offsetsHostPtr, countsHostPtr;
+
     if(DeviceExec)
     {
-      // Copy points to host memory.
-      // TODO: see if we support copying ArrayView<const T> -> Array<T>
-      ptsHost = HostPointArray(pts.size());
-      axom::copy(ptsHost.data(), pts.data(), sizeof(SpacePoint) * pts.size());
-      ptsHostPtr = ConstHostPointView(ptsHost.data(), ptsHost.size());
+      // Copy points and candidate intersections to host memory.
+      ptsHost = pts;
+      candidatesHost = candidates;
+      offsetsHost = offsets;
+      countsHost = counts;
+      // Set up views from intermediate host arrays
+      ptsHostPtr = ptsHost;
+      candidatesHostPtr = candidatesHost;
+      offsetsHostPtr = offsetsHost;
+      countsHostPtr = countsHost;
       // Allocate intermediate output buffers on the host side.
       outCellIdsHost.resize(pts.size());
       if(outIsoparametricCoords)
       {
         outIsoparHost.resize(pts.size());
       }
+      outCellIdsPtr = outCellIdsHost;
+      outIsoparPtr = outIsoparHost;
+    }
+    else
+    {
+      ptsHostPtr = pts;
+      candidatesHostPtr = candidates;
+      offsetsHostPtr = offsets;
+      countsHostPtr = counts;
```

Comment:
Are these copies necessary when `!DeviceExec`?

Reply:
In this case, these assignments are actually for creating host-side `ArrayView`s directly from the original `candidates/offsets/counts` arrays. In the device-side case a little further up, we do copy these arrays to an intermediate host-side array first, before initializing the views from the intermediate arrays.

Reply:
Thanks -- it's a bit difficult to see which ones are shallow `ArrayView` copies and which ones are deep `Array` copies. 
(I suspect it would be easier if you know to look for it :) )